### PR TITLE
Issue/2745 Add ability to mark/unmark a product as downloadable from product settings

### LINKED ISSUE #2745: Add/Edit Products M5: Downloadable files Step 2.1: Mark/unmark a product as downloadable
As part of the M5 release, we need to provide the option to change the `is_downloadable` flag of a product by adding an checkbox "Product Downloadable" to the product settings:

Screen Shot 2020-08-19 at 12 30 30 Screen Shot 2020-08-19 at 12 44 46

### PR DESCRIPTION
**Pull request by hichamboushaba**
Fixes #2745 

**Description**
This PR adds the ability to change the value of the flag `is_downloadable` for a product through the product settings.
List of changes:

- Adds a checkbox "Product Downloadable" in the product settings.
- Handle the visibility of the checkbox depending on the flag `PRODUCT_RELEASE_M4`.
- Created a custom layout for the modal UI, as the MaterialAlertDialog title's max lines is set to 2 lines, and AFAIK there is no easy way to customize this without a custom layout.
- Updates ProductDetailViewModel to reflect `isDownloadable` in changes checks, and to update its value when the product is being updated.

**Screenshots**
Screen Shot 2020-08-19 at 15 28 14
Screen Shot 2020-08-19 at 15 28 19

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/settings/ProductSettingsFragment.kt [hunk 1]
```diff
@@ -203,4 +213,26 @@ class ProductSettingsFragment : BaseProductFragment(), NavigationResult {
...
+            MaterialAlertDialogBuilder(requireActivity())
+                .setView(R.layout.dialog_uncheck_is_downloadable_warning)
+                .setPositiveButton(R.string.product_uncheck_is_downloadable_warning_yes_button) { _, _ ->
+                    updateProductDraft(false)
+                }
+                .setNegativeButton(R.string.product_uncheck_is_downloadable_warning_no_button) { _, _ ->
+                    checkBox.isChecked = true
+                }
+                .setCancelable(false)
+                .show()
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/settings/ProductSettingsFragment.kt:233]: I don't think we need to use a new UI here. There is a `CustomDiscardDialog` class that we use to display discard/confirmation dialogs across all of the product editing (and some of orders too I think). (Maybe change the name to make it more generic 😅 ).

**Review by anitaa1990 — COMMENTED** [review]
The code changes look good 👍 I left one code comment to perhaps reuse existing dialog instead of creating a new one. 

✅ Tested modifying the downloadable switch with existing files. 
✅ When I switch on the downloadable flag for a product that does not have any files, I don't see the downloadable section in the Product detail or in the Product detail bottom sheet. I assume this is because it's part of adding/uploading new downloadable files and not part of the scope of this PR.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/settings/ProductSettingsFragment.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/settings/ProductSettingsFragment.kt:233]: I just checked the `CustomDiscardDialog`, and internally it just uses the MaterialAlertDialogBuilder, so its title will have the same maximum number of lines (2), the custom layout is necessary in all cases, I think.

Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/settings/ProductSettingsFragment.kt:233]: Ahh, TIL that the max number of lines for the title is 2. Agree to leave it as is 👍

**Review by anitaa1990 — APPROVED** [review]
:shipit:

**Pull request merged**